app/helpers/advanced_messenger_helper.rb

```diff
@@ -74,29 +74,26 @@ def getUnreadNotificationsForCurrentUserCount()
     def getUsersReadStatus(read_by_users)
         read_by_users = JSON.parse(read_by_users)
         read_statuses = Hash.new
-        # For the moment the global users cache is implemented only for forum messages
-        # There is an open issue to implement it also for issues notes: https://redmine.xops-online.com/issues/33177
-        users_cache = @users_cache != nil ? @users_cache : Hash.new
         read_by_users.keys.each_with_index do |user_id, index|
```

Comment:
I'm not sure I see where you make the difference between `forum` messages and `issues`.

I mean if this is a common code, that how did it work for forum but not for notes?

Reply:
Inainte: pentru forum: @users_cache != nil ca il instantiam inainte, insa pentru issue, cum nu era instantiat se folosea un map local in aceasta functie instantiat cu Hash.new